lib/testutils/httpmultibin/httpmultibin.go

```diff
@@ -368,12 +367,14 @@ func NewHTTPMultiBin(t testing.TB) *HTTPMultiBin {
 		Dialer:          dialer,
 		HTTPTransport:   transport,
 		Context:         ctx,
-		Cleanup: func() {
```

Comment:
I think you should leave this as it was, and make the cleanup like this:
```go
t.Cleanup(func() {
    result.Cleanup()
}()
```

This will allow tests to modify the cleanup order, if they need to, by overwriting the `tb.Cleanup` lambda.

Reply:
Given that this hasn't been required so far I prefer to be added when it's needed